src/utils/retry_utils/src/retry_utils.c

```diff
@@ -186,14 +186,15 @@ int RetryUtils_GetRetryParamsMapSize()
 }
 
 void ReadRetryParamsArrayFromAgentConfigJson(
-    ADUC_Retriable_Operation_Context* context, JSON_Value* agentJsonValue, int retryParamsMapSize)
+    ADUC_Retriable_Operation_Context* context, JSON_Value* agentRetrySettingsJsonValue, int retryParamsMapSize)
 {
     const char* infoFormatString = "Failed to read '%s.%s' from agent config. Using default value (%d)";
 
     for (int i = 0; i < retryParamsMapSize; i++)
     {
         ADUC_Retry_Params* params = &context->retryParams[i];
-        JSON_Value* retryParams = json_object_dotget_value(json_object(agentJsonValue), s_retryParamsMap[i].name);
+        JSON_Value* retryParams =
```

Comment:
Might be overloading the function but should be using dothas_ to determine if we need to add the default configuration.

If the configuration has the value but it is invalid, we need to warn the customer instead of assuming it just doesn't exist.